mountpoint-s3/src/cli.rs

```diff
@@ -0,0 +1,250 @@
+        let _ = arg_matches.try_remove_one::<bool>("read_write");
+        CliArgs::from_arg_matches(&arg_matches).map_err(|e| anyhow!(e))
+    }
+
+    fn filter_option(option: &str) -> bool {
+        // Same as from https://github.com/s3fs-fuse/s3fs-fuse/blob/22ca6ba6ee9ffb27602046e6f4334ec4e9b0b7d7/src/s3fs.cpp#L5340-L5345
+        // and https://github.com/libfuse/sshfs/blob/ed0825440c48895b7e20cc1440bbafd8d9c88eb8/sshfs.c#L533-L538
+        ["auto", "noauto", "user", "nouser", "users", "_netdev"].contains(&option) || option.starts_with("x-")
+    }
+
+    fn split_commas(string: &str) -> anyhow::Result<Vec<String>> {
+        let mut unescaped = Vec::new();
+        let mut fragments = Vec::new();
+
+        let mut last_was_backslash = false;
+        let mut prev_idx = 0usize;
+        for (idx, char) in string.char_indices() {
+            if last_was_backslash {
+                match char {
+                    ',' | '"' | '\\' => {
+                        fragments.push(&string[prev_idx..idx - 1]);
+                        prev_idx = idx;
+                    }
+                    _ => {
+                        return Err(anyhow!(
+                            "Unexpected character after backslash escape - found {} at index {}",
+                            char,
+                            idx
+                        ));
+                    }
+                }
+            } else if char == ',' {
+                fragments.push(&string[prev_idx..idx]);
+                unescaped.push(fragments.join(""));
+                fragments.clear();
+                prev_idx = idx + 1;
+            } else if char == '"' {
+                return Err(anyhow!(
+                    "Unexpected '\"' found at index {} - perhaps you meant to escape it with \\",
+                    idx
+                ));
+            }
+            last_was_backslash = !last_was_backslash && char == '\\';
+        }
+        if last_was_backslash {
+            return Err(anyhow!("Unexpected end of string after '\\'"));
+        }
+        fragments.push(&string[prev_idx..]);
+        unescaped.push(fragments.join(""));
+        Ok(unescaped)
+    }
+}
+
+#[derive(Parser, Debug)]
+#[clap(disable_help_flag = true)]
+struct FsTabOptions {
+    #[clap(flatten)]
+    pub cli_args: CliArgsBase,
+    #[clap(
+        long="ro",
+        help = "Mount file system in read-only mode",
+        help_heading = MOUNT_OPTIONS_HEADER,
+        conflicts_with="read_write",
+    )]
+    pub read_only: bool,
+    #[clap(
+        long="rw",
+        help = "Mount file system in read-write mode",
+        help_heading = MOUNT_OPTIONS_HEADER,
+        conflicts_with="read_only",
+    )]
+    pub read_write: bool,
+    #[clap(long = "", default_value = "true", hide = true)]
+    pub foreground: bool,
+}
+
+pub fn get_cli_args() -> anyhow::Result<CliArgs> {
+    let is_fstab = env::args_os().len() == 5 && env::args_os().nth(3) == Some("-o".parse().unwrap());
```

Comment:
Not too sure about this. How do we report errors in each case? Do we want to show the help?

Reply:
Also what's `"-o".parse().unwrap()`?